projects/opendr_ws_2/src/opendr_perception/opendr_perception/skeleton_based_action_recognition_node.py

```diff
@@ -0,0 +1,243 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import torch
+import numpy as np
+
+import rclpy
+from rclpy.node import Node
+from std_msgs.msg import String
+from vision_msgs.msg import ObjectHypothesis
+from sensor_msgs.msg import Image as ROS_Image
+from opendr_ros2_bridge import ROS2Bridge
+from opendr_ros2_messages.msg import OpenDRPose2D
+
+from opendr.engine.data import Image
+from opendr.perception.pose_estimation import draw
+from opendr.perception.pose_estimation import LightweightOpenPoseLearner
+from opendr.perception.skeleton_based_action_recognition import SpatioTemporalGCNLearner
+from opendr.perception.skeleton_based_action_recognition import ProgressiveSpatioTemporalGCNLearner
+
+
+class SkeletonActionRecognitionNode(Node):
+
+    def __init__(self, input_rgb_image_topic="/usb_cam/image_raw",
+                 output_rgb_image_topic="/opendr/image_pose_annotated",
+                 pose_annotations_topic="/opendr/poses",
+                 output_category_topic="/opendr/skeleton_recognized_action",
+                 output_category_description_topic="/opendr/skeleton_recognized_action_description",
+                 device="cuda", model='stgcn'):
+        """
+        Creates a ROS2 Node for skeleton-based action recognition
+        :param input_rgb_image_topic: Topic from which we are reading the input image
+        :type input_rgb_image_topic: str
+        :param output_rgb_image_topic: Topic to which we are publishing the annotated image (if None, we are not publishing
+        annotated image)
+        :type output_rgb_image_topic: str
+        :param pose_annotations_topic: Topic to which we are publishing the annotations (if None, we are not publishing
+        annotated pose annotations)
+        :type pose_annotations_topic:  str
+        :param output_category_topic: Topic to which we are publishing the recognized action category info
+        (if None, we are not publishing the info)
+        :type output_category_topic: str
+        :param output_category_description_topic: Topic to which we are publishing the description of the recognized
+        action (if None, we are not publishing the description)
+        :type output_category_description_topic:  str
+        :param device: device on which we are running inference ('cpu' or 'cuda')
+        :type device: str
+        :param model:  model to use for skeleton-based action recognition.
+         (Options: 'stgcn', 'pstgcn')
+        :type model: str
+        """
+
```

Comment:
The call to super init is missing causing the node to not run.

```suggestion
        super().__init__('skeleton_based_action_recognition_node')
```